File: web-service/yacontest-helper/src/main/java/org/ilyatyamin/yacontesthelper/report/service/ReportService.java
package org.ilyatyamin.yacontesthelper.report.service;

import org.ilyatyamin.yacontesthelper.grades.dto.ContestSubmissionWithCode;
import org.ilyatyamin.yacontesthelper.report.dto.ReportRequest;

import java.util.List;
import java.util.Map;

public interface ReportService {
    byte[] createReport(ReportRequest request);
    String getMarkdownReport(ReportRequest request,
                             Map<String, Map<String, Double>> totalGrades,
                             Map<String, Double> submissionStatsOk,
                             List<ContestSubmissionWithCode> codeSubmissions);
}

File: web-service/yacontest-helper/src/main/java/org/ilyatyamin/yacontesthelper/report/service/ReportService.kt
package org.ilyatyamin.yacontesthelper.report.service

import org.ilyatyamin.yacontesthelper.grades.dto.ContestSubmissionWithCode
import org.ilyatyamin.yacontesthelper.report.dto.ReportRequest

interface ReportService {
    fun createReport(request: ReportRequest): ByteArray

    fun getMarkdownReport(
        request: ReportRequest,
        totalGrades: Map<String, Map<String, Double>?>,
        submissionStatsOk: Map<String, Double>,
        codeSubmissions: List<ContestSubmissionWithCode>
    ): String?
}
